lib/galaxy/webapps/galaxy/api/__init__.py

```diff
@@ -98,10 +98,12 @@ async def get_app_with_request_session() -> AsyncGenerator[StructuredApp, None]:
     app = get_app()
     request_id = request_context.data["X-Request-ID"]
     app.model.set_request_id(request_id)
+    app.install_model.set_request_id(request_id)
     try:
         yield app
     finally:
         app.model.unset_request_id(request_id)
+        app.install_model.unset_request_id(request_id)
```

Comment:
I noticed the FastAPI endpoint injects `ToolShedRepositoriesService` using `galaxy.webapps.galaxy.api.depends` but the latter [uses](https://github.com/galaxyproject/galaxy/blob/42829a5c9dc450883ef5dfc7608577e5a447fb61/lib/galaxy/webapps/galaxy/api/__init__.py#L121-L125) [`get_app()`](https://github.com/galaxyproject/galaxy/blob/42829a5c9dc450883ef5dfc7608577e5a447fb61/lib/galaxy/webapps/galaxy/api/__init__.py#L93-L94) rather than [`get_app_with_request_session()`](https://github.com/galaxyproject/galaxy/blob/42829a5c9dc450883ef5dfc7608577e5a447fb61/lib/galaxy/webapps/galaxy/api/__init__.py#L97-L104).

The two lines added here `app.install_model.set_request_id(request_id)` and `app.install_model.unset_request_id(request_id)` should solve the problem, but because of the above, they are not having any effect right?

Reply:
yes, this should all be fixed in https://github.com/galaxyproject/galaxy/pull/18629/commits/d20b3331f065db4782744ad27f20bc30a5db163a